File: flask_app/app/services/notification_service.py
from app.dao.notification_dao import NotificationDAO

class NotificationService:

    @staticmethod
    def get_all_for_user(user_id, page, per_page,filters=None):
        return NotificationDAO.get_all_for_user(user_id, page, per_page, filters)

    @staticmethod
    def create(data):
        return NotificationDAO.create(data)

    @staticmethod
    def mark_as_read(notification_id):
        return NotificationDAO.mark_as_read(notification_id)

    @staticmethod
    def get_by_id(notification_id):
        return NotificationDAO.get_by_id(notification_id)

    @staticmethod
    def mark_all_as_read(user_id):
        NotificationDAO.mark_all_as_read(user_id)

File: java_membership_api/src/main/java/com/example/membership_api/service/NotificationService.java
package com.example.membership_api.service;

import com.example.membership_api.dao.NotificationDAO;
import com.example.membership_api.model.Notifications;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
public class NotificationService {

    private final NotificationDAO notificationDAO;

    public NotificationService(NotificationDAO notificationDAO) {
        this.notificationDAO = notificationDAO;
    }

    public List<Notifications> getAllForUser(String userId, int page, int perPage, Map<String, Object> filters) {
        return notificationDAO.getAllForUser(userId, page, perPage, filters);
    }

    public long countForUser(String userId, Map<String, Object> filters) {
        return notificationDAO.countForUser(userId, filters);
    }

    public Notifications create(Notifications notification) {
        return notificationDAO.create(notification);
    }

    public Notifications markAsRead(String notificationId) {
        return notificationDAO.markAsRead(notificationId);
    }

    public Notifications getById(String notificationId) {
        return notificationDAO.getById(notificationId);
    }

    public void markAllAsRead(String userId) {
        notificationDAO.markAllAsRead(userId);
    }
}
